lib/fluent/plugin/out_papertrail.rb

```diff
@@ -1,30 +1,84 @@
-require 'syslog_protocol'
-require 'fluent/output'
-
 module Fluent
-  class Papertrail < Fluent::Output
-    attr_accessor :socket, :papertrail_host, :papertrail_port
+  class Papertrail < Fluent::BufferedOutput
+    attr_accessor :socket
+
+    # if left empty in fluent config these config_param's will error
+    config_param :papertrail_host, :string
+    config_param :papertrail_port, :integer
+    # overriding default flush_interval (60 sec) from Fluent::BufferedOutput
+    config_param :flush_interval, :time, default: 1
 
+    # register as 'papertrail' fluent plugin
     Fluent::Plugin.register_output('papertrail', self)
 
+    def initialize
+      super
+      require 'syslog_protocol'
+    end
+
     def configure(conf)
       super
-      @papertrail_host = conf["papertrail_host"]
-      @papertrail_port = conf["papertrail_port"]
-      @socket = UDPSocket.new
+      @socket = create_socket(@papertrail_host, @papertrail_port)
```

Comment:
These concerns seem to make more sense in the `initialize` function. As I read this, I'm not sure why I'm seeing both `initialize` and `configure`?

Reply:
There's probably no use for `initialize` if we move the require out. The super call to `configure` loads the fluent.conf file which is where the config_param's @papertrail_host and @papertrail_port get defined. Since those are needed to create the socket I put the initial socket creation in here but `send_to_papertrail` defines the socket if it doesn't exist anyway, so may be no need for this here at all.

Reply:
OK now I understand what's going on here re `configure` vs `initialize` thanks to @alexouzounis pointing me to the [plugin docs](https://docs.fluentd.org/v0.12/articles/plugin-development#writing-buffered-output-plugins)

Reply:
TL;DR: I think you can get away with removing the entire `initialize` method, but will need to keep `configure` basically as-written.

I'd remove the require from the `initialize` and see if you can get away with removing the whole method, as the `super` call only seems to do some weird metaprogramming bookkeeping. 

But the one in `configure` *does* seem to do a lot of state setup in the parent class, so I think you'll need to leave that one.

Reply:
Testing those changes out now. Will report back.